### PR TITLE
Always SDL_PumpEvents() in SDL_PeepEvents()

### PR DESCRIPTION
**Pull request by sulix**
SDL 1.2 always pumps events on a different thread on some systems, and some games will only call SDL_PeepEvents(), which doesn't otherwise pump events, and get stuck when they've consumed all the events.

Just pumping the event loop here simulates an event thread well enough for most things.

This is needed for the Loki port of Civilization: Call to Power (which is SDL 1.1, and has many other problems…) to not get stuck with a black screen on startup.

There's the possibility of putting this behind a hint (or trying to implement a more proper event thread) if it causes problems, but it hasn't seemed to so far.

### CODE REVIEW

**Comment by icculus** [comment]
Hmm, I don't have call to power...can you see if it tries to call SDL_Init with the 0x01000000 flag set? That would be SDL_INIT_EVENTTHREAD.

**Comment by sulix** [comment]
Ah… it seems to call SDL_Init twice. Once without SDL_INIT_EVENTTHREAD, once with:
```
SDL_Init(0x100100) /* SDL12_INIT_CDROM | SDL12_INIT_NOPARACHUTE */
SDL_Init(0x1100030) /* SDL12_INIT_AUDIO | SDL12_INIT_VIDEO | SDL_INIT_NOPARACHUTE | SDL_INIT_EVENTTHREAD */
```

**Comment by icculus** [comment]
I suspect that we should just fake the event thread the exact way your patch already does and not risk _actually_ threading it, but I don't know the full ramifications of that.

Let's make a note of it, remove the FIXME that says the event thread is unimplemented, and call it done until something in the wild turns out to be broken.

File: src/SDL12_compat.c [hunk 1]
```diff
@@ -2017,6 +2017,9 @@ SDL_InitSubSystem(Uint32 sdl12flags)
         if (Init12Video() < 0) {
             rc = -1;
         }
+
+        /* SDL_INIT_EVENTTHREAD takes effect when SDL_INIT_VIDEO is also set */
+        EventThreadEnabled = (sdl12flags & SDL12_INIT_EVENTTHREAD) ? SDL_TRUE : SDL_FALSE;
```
[the remark this answers is not in the document]
Comment (icculus) [reply src/SDL12_compat.c:2022]: Yes, for completeness, but in theory all the things that use this should only work when initialized anyhow.

[the remark this answers is not in the document]
Comment (sulix) [reply src/SDL12_compat.c:2022]: It shouldn't be necessary: SDL 1.2 really just updates the event thread state when initialising the video subsystem.

While it does shutdown the actual thread when SDL_VIDEO is quit, given that events shouldn't go through at all if the video system isn't enabled, there shouldn't be any difference.

(But if you'd rather this be reset there as well, I can add it…)

[the remark this answers is not in the document]
Comment (sezero) [reply src/SDL12_compat.c:2022]: Whatever you guys decide, I'm OK with it.

[the remark this answers is not in the document]
Comment (sulix) [reply src/SDL12_compat.c:2022]: I've just gone ahead and reset it in ``SDL_QuitSubsystem(SDL_INIT_VIDEO)``.

**Comment by sezero** [comment]
OK, let's get this in.

**Pull request merged**